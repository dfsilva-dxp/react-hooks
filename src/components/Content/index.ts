import styled, { css } from "styled-components";

const Content = styled.section`
  ${({ theme }) => css`
    width: 100%;
    height: auto;
    min-height: 100vh;
    padding: ${theme.spacings.medium} 0;

    display: flex;
    align-items: center;
    justify-content: center;
  `}
`;

export default Content;
