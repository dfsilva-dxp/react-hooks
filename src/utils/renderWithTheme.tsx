import { ReactNode } from "react";
import { render, RenderResult } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

import { MyTheme } from "../styles";

export const renderWithTheme = (children: ReactNode): RenderResult =>
  render(<ThemeProvider theme={MyTheme}>{children}</ThemeProvider>);
