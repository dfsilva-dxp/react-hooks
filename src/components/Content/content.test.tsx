import { screen } from "@testing-library/react";

import { renderWithTheme } from "../../utils/tests/renderWithTheme";

import Content from ".";

describe("<Content/>", () => {
  it("should render currectly", () => {
    const { container } = renderWithTheme(
      <Content>
        <h1>Content</h1>
      </Content>
    );

    const heading = screen.getByRole("heading", { name: /content/i });
    const parent = heading.parentElement;

    expect(parent).toHaveStyle({
      width: "100%",
      height: "auto",
      "min-height": "100vh",
      padding: "2rem 0",

      display: "flex",
      "align-items": "center",
      "justify-content": "center"
    });

    expect(container).toMatchSnapshot();
  });
});
