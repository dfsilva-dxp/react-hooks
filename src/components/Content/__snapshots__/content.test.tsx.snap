// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content/> should render currectly 1`] = `
.c0 {
  width: 100%;
  height: auto;
  min-height: 100vh;
  padding: 2rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div>
  <section
    class="c0"
  >
    <h1>
      Content
    </h1>
  </section>
</div>
`;
